### ISSUE TITLE
Problematische Autocomplete-Vorschläge in "Alphabetisch browsen"

### ISSUE THREAD

**Issue by coeh** [ubtue/tuefind#3082]
Beim alphabetischen Browsen nach Verfasser werden Autocomplete-Vorschläge angezeigt. Diese sind zum einen im Kontext des Browsens nach Verfasser*in wenig sinnvoll (enthalten Einträge aus dem Feld Titel). In diesem Beispiel habe ich einen Browsingvorgang nach Personen mit Nachname "Dietrich" gestartet und erhalte Vorschläge aus Titeln, die Personen mit Vorname "Dietrich" behandeln (da die Vornamen natürlich weiter vorne im Titel auftauchen):
![grafik](https://github.com/user-attachments/assets/c4cada51-32fc-48ab-8817-e6e6c29b959d)

Zum anderen erscheinen die Vorschläge - auch, nachdem man sie weggeklickt hat - jedesmal von neuem, wenn man auf die nächste Seite des Ergebnisses weiter blättert:
![grafik](https://github.com/user-attachments/assets/72a6e3c9-0ed0-47cf-a060-f4c29f2109dc)

Das Problem ist auf dem Livesystem von IxTheo reproduzierbar, also wohl keine direkte Folge des Upgrades. Als wir den Text zum Browsen für die neue Homepage geschrieben haben, war es aber noch nicht so (ca. vor 1-1,5 Jahren?). 

Ist im Kontext alphabetischen Browsens diese Funktion verzichtbar? Beim Browsen geht es ja gerade nicht um einen konkret gesuchten Titel o.ä., sondern ich möchte alle mit einem bestimmten Nachnamen, alle deren Name mit X beginnt... etc.

**Comment by NMagin** [comment]
Wie in Issue #3073 beschrieben tritt das Problem auch in RelBib auf. Grundsätzlich fände ich die dort beschriebene Einschränkung der Vorschläge auf Autor*innen (wie bei der Personensuche) sinnvoll, weil sie vielleicht hilfreich sein können, wenn man sich nicht mehr an den ganzen Namen oder die richtige Schreibweise erinnern kann. Aber grundsätzlich teile ich die Ansicht, dass die Vorschläge an dieser Stelle im Zweifel optional sind.

**Comment by jriedl** [comment]
Hier kommen unterschiedliche Aspekte zusammen: Die Auswahl der "Zielbereichs" der Vorschlagsfunktion: Dies wurde nun auf ptah so angepasst, dass es für die Kategorien außer dem Titel jetzt die richtige Kategorie gewählt werden sollte. Beim Titel ist das Problem, dass hier schon beim Import in den Index des alphabetisches Browsens eine komplexere Logik zum Einsatz kommt (Sortier- vs. Wertfelder + Normalisierung) und diese Suche sich offenbar momentan auch grundsätzlich nicht mehr wie intendiert verhält (die von @NMagin in #3074 beschriebenen Effekte). Dies betrifft aber offenbar schon die "offizielle" Vufind-Demo-Version (vgl. https://vufind.org/advanced_demo/Alphabrowse/Home): Eine Suche z.B. nach `Bibel`  führt mit und ohne Annahme des Vorschlags zu einer Ergebnisseite mit "Zygon":
![image](https://github.com/user-attachments/assets/87c8c4fb-7bc4-496e-abee-52d099dfa641)

 Im Tuefind-Kontext Stand wurde deshalb nun zumindest die Vorschlagsfunktion für die Titel des alphabetischen Browsens deaktiviert.

**Comment by coeh** [comment]
Vielen Dank für die Erläuterungen!

"Die Decke ist immer zu kurz" leuchtet ein. Die Hintergründe der übrigen Punkte sind mir jetzt auch klarer. Allerdings denke ich weiterhin, dass sie für Wissenschaftler*innen kontraintuitiv sind.

Grundsätzlich: Ich halte Autocomplete-Vorschläge beim alphabetischen Browsen wie oben erläutert nicht für notwendig, aber - solange sie niemanden verwirren - auch nicht per se für ein Problem. Der aktuelle Stand dürfte freilich die meisten Nutzenden eher irritieren oder (im Falle des immer neuen Anzeigens der Vorschläge auf jeder Seite) auch schlicht in praktischer Hinsicht nerven. Daher würde ich eher dazu tendieren, Autocomplete im Kontext des alphabetischen Browsens wegzulassen, da es dort  - anders als bei anderen Suchfunktionen - im Normalfall nicht benötigt wird. Was meinen die anderen? (@thefass @relhei @socheres @NMagin @JMeister01 )?

**Comment by relhei** [comment]
Ich stimme @coeh zu und befürworte das Weglassen der Autocomplete-Funktion beim alphabetischen Browsen.

**Comment by NMagin** [comment]
Wie oben schon geschrieben: Die Funktion kann grundsätzlich nützlich sein, aber wenn sie - wie aktuell - mehr verwirrt als hilft, sollte sie abgeschaltet werden.

**Comment by jriedl** [comment]
Da es bislang keine widersprechenden Meinungen gab, findet sich der Stand mit deaktivierter Vorschlagfunktion nun auf ptah.

**Comment by coeh** [comment]
Ich habe mit @thefass gesprochen und er ist auch einverstanden, dass diese Funktion im Kontext des alphabetischen Browsens abgeschaltet wird.

Ich habe außerdem eben getestet:
- Alphabetisch browsen nach Thema funktioniert wie gewünscht
- Alphabetisch browsen nach Verfasser funktioniert wie gewünscht

Alphabetisch browsen nach Titel funktioniert auf ptah allerdings gerade nicht richtig: Eingabe "Jenseits" ergibt Vorschläge mit A und die Meldung "Ihr Treffer würde hier aufgelistet":
![grafik](https://github.com/user-attachments/assets/37066afe-1462-4d9e-9fac-105e1684886f)

Wir haben aber definitiv diverse Titel, die mit "Jenseits" beginnen, wie man bei einer normalen Suche sieht:
![grafik](https://github.com/user-attachments/assets/6bf3ad38-aaa2-48f6-b6b9-c0a61881101b)

**Comment by jriedl** [comment]
> Alphabetisch browsen nach Titel funktioniert auf ptah allerdings gerade nicht richtig: Eingabe "Jenseits" ergibt Vorschläge mit A und die Meldung "Ihr Treffer würde hier aufgelistet":

Das ist die oben beschriebene, bereits das offizielle VuFind (s.u.) und auch die aktuellen Livesysteme (Suche nach "Jenseits" dort liefert ebenfalls bereits das oben beschriebene Ergebnis) betreffende "kaputte" Titelsuche:

> Beim Titel ist das Problem, dass hier schon beim Import in den Index des alphabetisches Browsens eine komplexere Logik zum Einsatz kommt (Sortier- vs. Wertfelder + Normalisierung) und diese Suche sich offenbar momentan auch grundsätzlich nicht mehr wie intendiert verhält (die von @NMagin in #3074 beschriebenen Effekte). Dies betrifft aber offenbar schon die "offizielle" Vufind-Demo-Version (vgl. https://vufind.org/advanced_demo/Alphabrowse/Home): Eine Suche z.B. nach `Bibel` führt mit und ohne Annahme des Vorschlags zu einer Ergebnisseite mit "Zygon":

Die möglichen weiteren Vorgehensweisen (Abwarten der Behebung in der offiziellen Version, Versuch einer Behebung auf unserer Seite, Deaktivierung der Titelsuche im Alphabetischen Browsen) müssten hier abgesprochen werden.

**Comment by jriedl** [comment]
Die Fehlerbilder zwischen dem offiziellen VuFind und unserem System waren zwar ähnlich, hatten aber, wie sich nun aufgeklärt hat, unterschiedliche Ursachen. Die seit einem weiter zurückliegenden Update bestehende Diskrepanz zwischen den Normalisierungsstrategien bei Import und Suche in der alphabetischen Suche, die das Problem mit den Titeln bei TueFind verursacht hat, ist nun behoben. Aktueller Stand auf ptah und sobek.
